webdriver/get-title.py

```diff
@@ -0,0 +1,268 @@
+    session.url = inline("<title>Foobar</title><h2>Hello</h2>")
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+    assert_success(result, "Foobar")
+
+# [...]
+# 3. Let title be the initial value of the title IDL attribute of the current
+#    top-level browsing context's active document.
+# 4. Return success with data title.
+# [...]
+# The title attribute must, on getting, run the following algorithm:
+# [...]
+# 2. Otherwise, let value be the child text content of the title element [...]
+#
+#    The title element of a document is the first title element in the document
+#    (in tree order), if there is one, or null otherwise.
+#
+# [...]
+# 4. Return value.
+def test_title_with_duplicate_element(session):
+    session.url = inline("<title>First title</title><title>Second title</title>")
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+
+    assert_success(result, "First title")
+
+# [...]
+# 3. Let title be the initial value of the title IDL attribute of the current
+#    top-level browsing context's active document.
+# 4. Return success with data title.
+# [...]
+# The title attribute must, on getting, run the following algorithm:
+# [...]
+# 2. Otherwise, let value be the child text content of the title element, or
+#    the empty string if the title element is null.
+# [...]
+# 4. Return value.
+def test_title_without_element(session):
+    session.url = inline("<h2>Hello</h2>")
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+
+    assert_success(result, "")
+
+# [...]
+# 3. Let title be the initial value of the title IDL attribute of the current
+#    top-level browsing context's active document.
+# 4. Return success with data title.
+def test_title_after_modification(session):
+    session.url = inline("<title>Initial</title><h2>Hello</h2>")
+    session.execute_script("document.title = 'updated';")
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+
+    assert_success(result, "updated")
+
+# [...]
+# 3. Let title be the initial value of the title IDL attribute of the current
+#    top-level browsing context's active document.
+# 4. Return success with data title.
+# [...]
+# The title attribute must, on getting, run the following algorithm:
+# [...]
+# 2. Otherwise, let value be the child text content of the title element [...]
+# 3. Strip and collapse ASCII whitespace in value.
+# 4. Return value.
+def test_title_strip_and_collapse(session):
+    document = "<title>   a b\tc\nd\t \n e\t\n </title><h2>Hello</h2>"
+    session.url = inline(document)
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+
+    assert_success(result, "a b c d e")
```

Comment:
Can you say why? At first glance, using `document.title` for this would seem to introduce a dependency on functionality that is not directly related to the behavior under test. E.g. if `document.title` were to return a different result, then we would want to write a test in HTML for that.